File: algorithms/bubble_sort.py
from random import randint

array = [randint(0, 20) for i in range(0, 20)]

print(array)


def bubble_sort(array):
    for i in range(len(array)-1):
        swapped = False
        for j in range(len(array)-i-1):
            if array[j] > array[j+1]:
                temp = array[j]
                array[j] = array[j+1]
                array[j+1] = temp


print(len(array))
bubble_sort(array)
print(array)
print(len(array))

File: algorithms/bubble_sort.c
#include <stdio.h>
#include <stdlib.h>

void print(int *list, int size)
{
    for (int i = 0; i < size; i++)
        printf("%d\t", list[i]);
}

void bubble_sort(int *list, int size)
{
    for (int i = 0; i < size; i++)
    {
        for (int j = 0; j < size - i - 1; j++)
        {
            if (list[j] > list[j + 1])
            {
                int *temp = list[j];
                list[j] = list[j + 1];
                list[j + 1] = temp;
            }
        }
    }
}

int main()
{
    int l[10];
    for (int i = 0; i < 10; i++)
    {
        l[i] = rand();
    }
    print(l, 10);
    printf("\n");
    bubble_sort(l, 10);
    print(l, 10);
    return 0;
}